lib/internal/mime.js

```diff
@@ -80,29 +82,49 @@ function parseTypeAndSubtype(str) {
   };
 }
 
-const EQUALS_SEMICOLON_OR_END = /[;=]|$/;
-const QUOTED_VALUE_PATTERN = /^(?:([\\]$)|[\\][\s\S]|[^"])*(?:(")|$)/u;
-
-function removeBackslashes(str) {
-  let ret = '';
-  // We stop at str.length - 1 because we want to look ahead one character.
-  let i;
-  for (i = 0; i < str.length - 1; i++) {
-    const c = str[i];
-    if (c === '\\') {
-      i++;
-      ret += str[i];
+function isSemicolonOrEquals(char) {
+  return char === SEMICOLON || char === EQUALS;
+}
+
+function isQuoteOrBackslash(char) {
+  return char === QUOTE || char === BACKSLASH;
+}
+
+function isNotWhiteSpace(char) {
+  return !RegExpPrototypeTest(WHITESPACE_REGEX, char);
+}
+
+function moveUntil(str, predicate, marker, { toASCII, removeBackslashes } = { toASCII: false, removeBackslashes: false }) {
+  const start = marker.position;
+  const length = str.length;
+  let index = start;
+  let result = '';
+
+  while(predicate(str[index]) === false && index < length) {
+    let char = str[index];
+    if (removeBackslashes && char === '\\') {
+      // jump to the next char
+      index++
+      char = str[index];
+    }
+
+    if (toASCII) {
+      const asCharCode = StringPrototypeCharCodeAt(char)
+      result += asCharCode >= ASCII_LOWER_LOWERBOUND && asCharCode <= ASCII_LOWER_UPPERBOUND ?
+      StringPrototypeToLowerCase(char) :
+      char;
     } else {
-      ret += c;
+      result += char;
```

Comment:
Potentially, if you reserve the length of `result` and trim depending on the length of the input it could run faster.

Reply:
It wouldn't have the same effect to construct a new string on every operation.

Please correct me if I misunderstood, but by doing that we might lose the individual char parse into ASCII, translating into a second iteration of string to achieve it (which was the initial approach of this step before my changes).

As I said, I may be misunderstood, so happy to hear your thoughts 🙂